geoengine/workflow.py

```diff
@@ -464,12 +465,18 @@ def save_as_dataset(
 
         return Task(TaskId.from_response(response))
 
+    # pylint: disable=too-many-locals
     async def raster_stream(
-            self,
-            query_rectangle: QueryRectangle,
-            open_timeout: int = 60) -> AsyncIterator[RasterTile2D]:
+        self,
+        query_rectangle: QueryRectangle,
+        open_timeout: int = 60,
+        bands: Optional[List[int]] = None  # TODO: move into query rectangle?
```

Comment:
Can't you say

```suggestion
        bands: List[int] = [0]  # TODO: move into query rectangle?
```

instead of allowing none and then instantly replacing it with Some([0])?

Reply:
that's what I initially did, but actually I lreaned form the linter that's an antipattern in Python. Because lists are mutable, basically if you change the list you change the default argument for all future calls of the function. So one should not use lists or other mutable objects as default values.